tests/producer_split_at.rs

```diff
@@ -224,6 +224,12 @@ fn enumerate() {
     check(&v, || (0..10).into_par_iter().enumerate());
 }
 
+#[test]
+fn step_by() {
+    let v: Vec<_> = (0..10).step_by(2).collect();
```

Comment:
Can you also add a case where the step does not evenly divide the total length? I think this will expose the issue I noted in `split_at`.